spec/graphql/schema/mutation_spec.rb

```diff
@@ -353,4 +353,87 @@ def self.resolve_type(abs_type, obj, ctx)
       assert_equal ["1", "1", "1"], result.map { |r| r["data"].first.last["id"] }
     end
   end
+
+  class LogArgumentsExtension < GraphQL::Schema::FieldExtension
+    extras [:ast_node]
+    def resolve(object:, arguments:, context:)
+      # I would like to find a way to log the arguments with the following changes:
+      # 1. Use the camelcase version
+      # 2. Use the raw inputs, not the Ruby parsed ones
+      ast_node = arguments[:ast_node]
+      formatted_args = {}
+      ast_node.arguments.each do |argument_node|
+        formatted_args[argument_node.name] = format_argument(context, argument_node.value)
+      end
+      context[:logged_arguments] = formatted_args
```

Comment:
The GraphQL spec allows clients to run multiple mutations in a single operation. If that happens, then subsequent assignments to `context[:logged_arguments]` will clobber previous ones. 

If you need to avoid that, you could do something like 

```ruby
logged_arguments_by_path = context[:logged_arguments] ||= {} 
logged_arguments_by_path[context.current_path] = formatted_args
```

Reply:
👍🏻 thank you! In practice, I was planning to emit a log and not store in `context`, just did this for writing this test easily

Do you think there is any place in the gem for having a utility to support this kind of log formatting? I am guessing not - it's quite easy to take the `ast_node` and I am adding a few more cases (Hash, TrueClass, FalseClass, and anything else that CI finds for me) but wanted to ask

Reply:
I don't think so ... we can point people here if it comes up again, and if it turns out to be widely used, we can add it later.